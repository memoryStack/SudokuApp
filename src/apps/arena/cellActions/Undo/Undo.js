import React from 'react'

import PropTypes from 'prop-types'

import _noop from '@lodash/noop'

import { UndoIcon } from '@resources/svgIcons/undo'

import Text from '@ui/atoms/Text'

import { Touchable } from '../../../components/Touchable'

import { styles, INACTIVE_ICON_FILL } from '../style'

const Undo = ({ iconBoxSize, onClick, ...rest }) => (
    <Touchable style={styles.actionContainer} onPress={onClick} {...rest}>
        <UndoIcon iconBoxSize={iconBoxSize} fill={INACTIVE_ICON_FILL} />
        <Text style={styles.actionText}>Undo</Text>
    </Touchable>
)

export default React.memo(Undo)

Undo.propTypes = {
    iconBoxSize: PropTypes.number,
    onClick: PropTypes.func,
}

Undo.defaultProps = {
    iconBoxSize: 40,
    onClick: _noop,
}
