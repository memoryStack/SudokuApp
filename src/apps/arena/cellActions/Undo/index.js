export { default } from './Undo'
export * from './undo.constants'
