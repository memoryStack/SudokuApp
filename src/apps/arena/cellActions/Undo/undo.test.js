import * as React from 'react'

import {
    fireEvent, render, screen,
} from '@utils/testing/testingLibrary'

import Undo from './Undo'

test('snapshot for  Undo', () => {
    const tree = render(<Undo iconBoxSize={50} />).toJSON()
    expect(tree).toMatchSnapshot()
})

describe('Undo', () => {
    test('will call handler on click', () => {
        const onClick = jest.fn()
        render(<Undo iconBoxSize={50} onClick={onClick} />)

        fireEvent.press(screen.getByText('Undo'))

        expect(onClick).toHaveBeenCalledTimes(1)
    })

    test('if disabled, will not call handler on click', () => {
        const onClick = jest.fn()
        render(<Undo iconBoxSize={50} onClick={onClick} disabled />)

        fireEvent.press(screen.getByText('Undo'))

        expect(onClick).not.toHaveBeenCalled()
    })
})
