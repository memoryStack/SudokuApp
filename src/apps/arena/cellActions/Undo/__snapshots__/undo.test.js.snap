// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for  Undo 1`] = `
<View
  accessibilityRole={null}
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={null}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID=""
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={46}
    bbWidth={46}
    focusable={false}
    height={46}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 46,
          "width": 46,
        },
      ]
    }
    vbHeight={512}
    vbWidth={512}
    width={46}
  >
    <RNSVGGroup>
      <RNSVGPath
        d="M212.333 224.333H12c-6.627 0-12-5.373-12-12V12C0 5.373 5.373 0 12 0h48c6.627 0 12 5.373 12 12v78.112C117.773 39.279 184.26 7.47 258.175 8.007c136.906.994 246.448 111.623 246.157 248.532C504.041 393.258 393.12 504 256.333 504c-64.089 0-122.496-24.313-166.51-64.215-5.099-4.622-5.334-12.554-.467-17.42l33.967-33.967c4.474-4.474 11.662-4.717 16.401-.525C170.76 415.336 211.58 432 256.333 432c97.268 0 176-78.716 176-176 0-97.267-78.716-176-176-176-58.496 0-110.28 28.476-142.274 72.333h98.274c6.627 0 12 5.373 12 12v48c0 6.627-5.373 12-12 12z"
        fill={4286545791}
        propList={
          Array [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <Text
    style={
      Object {
        "color": "rgb(127, 127, 127)",
        "fontFamily": "OpenSans",
        "fontSize": 16,
        "fontWeight": "400",
        "includeFontPadding": false,
        "letterSpacing": 0.5,
        "lineHeight": 24,
        "marginTop": 8,
        "textAlignVertical": "center",
      }
    }
    testID=""
  >
    Undo
  </Text>
</View>
`;
